core/src/main/scala/org/apache/spark/internal/io/HadoopMapReduceCommitProtocol.scala

```diff
@@ -157,6 +159,7 @@ class HadoopMapReduceCommitProtocol(
     // Include a UUID here to prevent file collisions for one task writing to different dirs.
     // In principle we could include hash(absoluteDir) instead but this is simpler.
     val tmpOutputPath = new Path(stagingDir, UUID.randomUUID().toString() + "-" + filename).toString
+    stagingDirExists = true
```

Comment:
Updated the checking in `commitJob`, but since result can't be retrieved in `abortJob`, so check before delete is implemented in `abortJob`.
@cloud-fan Could you help to check?